repl/src/main/scala/com/cloudera/livy/repl/PythonInterpreter.scala

```diff
@@ -124,6 +132,67 @@ object PythonInterpreter extends Logging {
 
     file
   }
+
+  private def initiatePy4jCallbackGateway(server: GatewayServer): PySparkJobProcessor = {
+    val f = server.getClass.getDeclaredField("gateway")
+    f.setAccessible(true)
+    val gateway = f.get(server).asInstanceOf[Gateway]
+    val command: String = "f" + Protocol.ENTRY_POINT_OBJECT_ID + ";" +
+      "com.cloudera.livy.repl.PySparkJobProcessor"
+    getPythonProxy(command, gateway).asInstanceOf[PySparkJobProcessor]
+  }
+
+  // This method is a hack to get around the classLoader issues faced in py4j 0.8.2.1 for
+  // dynamically adding jars to the driver. The change is to use the context classLoader instead
+  // of the system classLoader when initiating a new Proxy instance
+  // ISSUE - https://issues.apache.org/jira/browse/SPARK-6047
+  // FIX - https://github.com/bartdag/py4j/pull/196
+  private def getPythonProxy(commandPart: String, gateway: Gateway): Any = {
+    val proxyString = commandPart.substring(1, commandPart.length)
+    val parts = proxyString.split(";")
+    val length: Int = parts.length
+    val interfaces = ArrayBuffer.fill[Class[_]](length - 1){ null }
+    if (length < 2) {
+      throw new Py4JException("Invalid Python Proxy.")
+    }
+    else {
+      var proxy: Int = 1
+      while (proxy < length) {
+        {
+          try {
+            interfaces(proxy - 1) = Class.forName(parts(proxy))
+            if (!interfaces(proxy - 1).isInterface) {
+              throw new Py4JException("This class " + parts(proxy) +
+                " is not an interface and cannot be used as a Python Proxy.")
+            }
+          }
+          catch {
+            case exception: ClassNotFoundException => {
+              throw new Py4JException("Invalid interface name: " + parts(proxy))
+            }
+          }
+        }
+        proxy += 1;
+      }
+      Proxy.newProxyInstance(Thread.currentThread.getContextClassLoader,
+        interfaces.toArray, new PythonProxyHandler(parts(0), gateway.getCallbackClient, gateway))
+    }
+  }
+
+  def createWrapper(driver: ReplDriver, msg: BaseProtocol.BypassJobRequest): BypassJobWrapper = {
```

Comment:
No; add these as methods of the class, not the object; then you don't need the `pysparkJobProcessor` field in the object (it's always sketchy to store things in static fields).